xheap.py

```diff
@@ -114,31 +114,22 @@ def push(self, item):
         _siftdown(self, 0, len(self)-1)
 
     def pop(self, index=None):
+        index = index or 0
+        if index < 0:
+            index += len(self)
+        if index < 0:
+            raise IndexError
         return_item = self._pop(index)
         del self._indexes[return_item]
         return return_item
 
     def _pop(self, index):
-        if index:
-            last_item = super(Heap, self).pop()
-            if index == len(self):
-                return_value = last_item
-            else:
-                return_value = self[index]
-                self[index] = last_item
-                if self[(index - 1) >> 1] < last_item:
```

Comment:
wow, i never bothered to see the _siftup code of heapq module. i really have no idea why there not breaking when both of children's values are valid w.r.t heap[pos]
let me ask about it in the python-list and come back here.